src/problems/sudoku.clj

```diff
@@ -0,0 +1,124 @@
+(ns problems.sudoku)
+;;OH BOY KIDS HERE WE GO
+(defn distinct?
+  "Returns true if no two of the arguments are ="
+  [& more]
+  (let [old (count more)
+        set (into #{} more)
+        new (count set)]
+    (= new old)))
+
+;;## validation top level ##
+(defn every-element-in-vec-distinct?
+ "Returns True if every element in every vector is unique or nil.
+  Returns False otherwise."
+  [vecs]
+ (every? identity (map #(apply distinct? %) vecs)))
+
+(def every-vector-valid? (memoize every-element-in-vec-distinct?))
+
+;;## validating 3x3 blocks ##
+(defn hop [adjusted-n k chunk]
+  "Given the first number in the block (n), and the offset (k), returns the 3
+  numbers that make up a column in a sudoku block."
+  (take-nth 9 (drop (+ adjusted-n k) chunk)))
+
+(defn hop-block [chunk n]
+  "Gets the three columns in a 3x3 sudoku block."
+  (let [adjusted-n (* n 3)] ;; the upper left hand number in a sudoku block.
+    (into [] (concat (hop adjusted-n 0 chunk)
+                     (hop adjusted-n 1 chunk)
+                     (hop adjusted-n 2 chunk)))))
+
+(defn get-blocks [chunk]
```

Comment:
I'm a little confused about why you're acting on 3x9 chunks. I can imagine that it might make sense, but I don't really get it from looking at the code.